20-git.md

```diff
@@ -0,0 +1,139 @@
+version are easily accessible. This reduces the need for ridiculous
+`_version5.23_final` filename appendices.
+
+Another common challenges that every physics student has to face in their time
+at university is collaborative work with fellow students. Let's take a lab
+course for example: You gather data, analyse it, and write a report about your
+findings. There's a lot to be versioned: your data (although it will not change
+after it's been recorded), your analysis programs, and the LaTeX source of your
+report.
+
+ * Superiority over exchanging files via email
+ * Superiority over exchanging files via Dropbox
+ * Two sides: Collaboration and backup/history
+
+
+## Git-ing started
+Before you can use Git, you need to configure it. This is necessary, as Git
+will use your name and email address to reference the changes you made to a
+project. Run the following commands in your terminal substituting your name and
+email address accordingly:
+
+```bash
+git config --global user.name "Surname Name"
+git config --global user.email "surname.name@cern.ch"
+```
+
+You can also set other options like for example coloured output with the same
+command:
+
+```bash
+git config --global color.ui auto
+```
+
+You can view all your settings by running `cat ~/.gitconfig`. You can also edit
+this file by hand if you know what you're doing. Extensive documentation can be
+found on the `git-config` manpage (run `man git-config`) or in the [online
+documentation](https://git-scm.com/docs/git-config).
+ 
+
+## Concepts
+
+
+## Creating a repository
+
+
+## Managing changes
+
+
+## Collaborating
+One important feature is the `.gitignore` file. It defines a list of files
+(wildcards are possible) that should always be ignored by Git for the current
+repository. For example, if you are collaborating on a paper in LaTeX, you will
+want to put your `.tex` files under version control but not any of the
+intermediate files produced by your LaTeX compilation. In your repository's root
+folder, create a file called `.gitinore` and fill it with the following lines:
+
+```
+*.acn
+*.acr
+*.alg
+*.aux
+*.bbl
+*.blg
+*.dvi
+*.glg
+*.glo
+*.gls
+*.idx
+*.ilg
+*.ind
+*.ist
+*.lof
+*.log
+*.lot
+*.maf
+*.mtc
+*.mtc1
+*.out
+*.synctex.gz
+*.toc
+```
+
+Please not that this list does not include `.pdf` files as you will probably add
+figures as PDF files. You will want to add your resulting `.pdf` file to that
+list above to keep the (dynamic) binary file from being put under version
+control.
+
+For files that are specific to your system (like `.DS_Store` files on OS X or
+`.swp` files if you are using Vim) you can globally define a list of file types
+that should be ignored by Git. Those files have nothing to do with your project
+and are specific to _your_ system. Your collaborator might be using Emacs (shame
+on them!) on Linux and therefore not care about those files. Create a new file
+in your home directory called `.gitignore` (or choose a name of your liking) and
```

Comment:
Remove the stuff in parentheses.  A true "power user" might want a different name for some bizarre reason, to every one else it's just confusing.